sdk/identity/identity/src/credentials/interactiveBrowserCredential.ts

```diff
@@ -104,7 +118,41 @@ export class InteractiveBrowserCredential implements TokenCredential {
   ): Promise<AccessToken | null> {
     const scopeArray = typeof scopes === "object" ? scopes : [scopes];
 
-    return this.acquireTokenFromBrowser(scopeArray);
+    if (this.authenticationRecord && this.persistenceEnabled) {
+      try {
+        return this.acquireTokenFromCache();
+      } catch (e) {
+        if (e instanceof AuthenticationRequired) {
```

Comment:
Do you have a link to a preferred pattern, by chance?

Reply:
If you set `this.name = "YourErrorType"` in your custom error constructor, you can check if `e.name === "YourErrorType"`. It's not necessarily *better*, but duck typing on a discriminant seems a little better than relying on two packages using the same constructor function for an error.

Reply:
It'll also print your custom error's class name when toString is called, along with message.

Reply:
Luckily, we have full control here, since we both define and are the only ones throwing this error. I think I'll leave it as-is for now, but something we might want to throw into the next JS/TS team discussion around preferred coding patterns.